chemprop/cli/utils/parsing.py

```diff
@@ -213,17 +216,42 @@ def build_data_from_files(
     smiss, rxnss, Y, weights, lt_mask, gt_mask = parse_csv(
         p_data, smiles_cols, rxn_cols, target_cols, ignore_cols, weight_col, bounded, no_header_row
     )
-    X_f = np.load(p_features) if p_features else None
-    V_fs = np.load(p_atom_feats, allow_pickle=True) if p_atom_feats else None
-    E_fs = np.load(p_bond_feats, allow_pickle=True) if p_bond_feats else None
-    V_ds = np.load(p_atom_descs, allow_pickle=True) if p_atom_descs else None
+    n_molecules = len(list(zip(*smiss)))
+
+    X_fs = load_input_features(p_features, n_molecules, feature="X_f")
+    V_fss = load_input_features(p_atom_feats, n_molecules, feature="V_f")
+    E_fss = load_input_features(p_bond_feats, n_molecules, feature="E_f")
+    V_dss = load_input_features(p_atom_descs, n_molecules, feature="V_d")
 
     mol_data, rxn_data = make_datapoints(
-        smiss, rxnss, Y, weights, lt_mask, gt_mask, X_f, V_fs, E_fs, V_ds, **featurization_kwargs
+        smiss, rxnss, Y, weights, lt_mask, gt_mask, X_fs, V_fss, E_fss, V_dss, **featurization_kwargs
     )
 
     return mol_data + rxn_data
 
+def load_input_features(paths, n_molecules, feature):
+    if paths is None:
+        return None
+
+    match feature:
+        case "X_f":
+
+            features = []
+            for mol_idx in range(n_molecules):
+                path = paths # TODO: currently only supports a single path
+                loaded_feature = np.load(path)
+                loaded_feature = loaded_feature["arr_0"]
+                features.append(loaded_feature)
+
+        case _:
+
+            features = []
+            for mol_idx in range(n_molecules):
+                path = paths # TODO: currently only supports a single path
+                loaded_feature = np.load(path)
+                loaded_feature = [loaded_feature[f"arr_{i}"] for i in range(len(loaded_feature))]
```

Comment:
Are we allowing for different additional features and descriptors to be in the same file? I remember there was a discussion but I don't remember what we decided on. If multiple feature or descriptor types are given in a single file, then if I'm understanding this correctly, features will be erroneously loaded more than once. I think separating out molecule, atom, and bond information is fine to expect. I'm not sure about atom features and descriptors though. v1 made the users choose between using either atom features/descriptors and so they could be in the same file, but we'd encounter an issue if a user did want to use both atom features and atom descriptors and had them in the same input file. I think this might be tricky to do and so it could be clearer to say that they are expected to be in different files.

Reply:
I'm not exactly sure what you meant. Do you mean the discussion whether we will have one central input file that has all the information for the smiles, targets, additional atom features, and additional atom descriptors, etc.? That is outside the scope of this PR.

This PR only adds supports for `.npz` format for atom features, descriptors, etc., as separate files, and each file expects them to be in the same sequence as the input smiles.